### PR TITLE
Remove list pages

### PR DESCRIPTION
**Pull request by edmundito**
## Change description

This removes the list pages that are no longer used now that we've switched to the model-centric UI. Pages removed include:

- Sources
- Properties
- Destinations
- Groups

It also removes the add sample record page and replaces it with the add sample record modal used by the sample record card.

### Development

- [x] Application changes have been tested appropriately

### Impact

- [ ] Code follows company security practices and guidelines
- [ ] Security impact of change has been considered
- [ ] Performance impact of change has been considered
- [ ] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [x] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

File: ui/ui-components/pages/model/[modelId]/destination/[destinationId]/edit.tsx [hunk 1]
```diff
@@ -115,8 +115,8 @@ export default function Page(props) {
       );
       if (success) {
         router.push(
-          "/model/[modelId]/destinations",
-          `/model/${destination.modelId}/destinations`
+          "/model/[modelId]/overview",
+          `/model/${destination.modelId}/overview`
         );
```
Comment (evantahler) [inline ui/ui-components/pages/model/[modelId]/destination/[destinationId]/edit.tsx:120]: 👍 deletes go to the model overview

File: ui/ui-components/pages/model/[modelId]/destinations.tsx [hunk 2]
```diff
@@ -1,206 +0,0 @@
...
-import LoadingTable from "../../../components/LoadingTable";
-import AppIcon from "../../../components/AppIcon";
-import StateBadge from "../../../components/badges/StateBadge";
-import { Models, Actions } from "../../../utils/apiData";
-import { formatTimestamp } from "../../../utils/formatTimestamp";
-import { ErrorHandler } from "../../../utils/errorHandler";
-import LinkButton from "../../../components/LinkButton";
-import { grouparooUiEdition } from "../../../utils/uiEdition";
-import { formatName } from "../../../utils/formatName";
-import DestinationCollectionLink from "../../../components/destination/DestinationCollectionLink";
```
Comment (evantahler) [inline ui/ui-components/pages/model/[modelId]/destinations.tsx:21]: I wonder if this component can be removed

File: ui/ui-config/pages/model/[modelId]/record/new.tsx [hunk 3]
```diff
@@ -1,122 +0,0 @@
-import Head from "next/head";
```
Comment (evantahler) [inline ui/ui-config/pages/model/[modelId]/record/new.tsx:1]: 👍 the only way to add a record manually is when in ui-config, and using the modal now

**Review by evantahler — APPROVED** [review]
Looking good!

File: ui/ui-components/pages/model/[modelId]/destinations.tsx (hunk 2 above)
Comment (edmundito) [reply ui/ui-components/pages/model/[modelId]/destinations.tsx:21]: I reviewed a lot of these components. I created this component to share between the model overview and this page. But now that the page is removed the overview still uses it

**Pull request merged**